roadmap/propagate-rust-panic-through-native-frame.md

```diff
@@ -0,0 +1,48 @@
+# Propagate Rust panic through native frames
+
+## Summary
+
+These roadmap items refer to when a panic is raised in a Rust function
+that is defined with `"C unwind"` ABI, as would happen if C or C++
+code invoked the following function:
+
+```rust
+extern "C unwind" fn example() {
+    panic!("Uh oh");
+}
+```
+
+In this case, the stack would look like one of the two scenarios:
+
+```
+     ...
+      |
+[Native frame]
+      |                           ^
+      | (calls)                   | (unwinding
+      v                           |  goes this
+[Rust function `example`]         |  way)
+      |                           |
+      +--- rust function panics --+
+```
+
+We can distinguish three cases:
+
+* The native frame has **no destructors**. An example of this might be a C function.
+    * The precise meaning of "no destructors" is defined in terms of the native ABI.
+    * We need to define how the Rust panic propagates across such frames.
+* The native frame **has destructors** it would like to execute. These could come from C++
+  RAII, Java finally blocks, etc.
```

Comment:
This would only apply to Java code that is compiled rather than running in the JVM, right? It seems like an odd example, since (as far as I know) it's not possible for native exceptions to propagate into JVM stack frames (except by through the VM code itself).

Reply:
Sorry, yeah, weird example, I was thinking of [gcj](https://en.wikipedia.org/wiki/GNU_Compiler_for_Java) but I guess that's not maintained anymore.

Reply:
I'm sure there's _some_ compiled language with `finally` blocks...I just don't know what it is! 😆 I'm guessing `rustc` targeting MSVC can't generate CLR code...